eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/io/BackPropagatedRSTargetReaderConfiguration.java

```diff
@@ -21,22 +21,36 @@
 import org.springframework.data.mongodb.core.MongoTemplate;
 
 import uk.ac.ebi.eva.accession.clustering.batch.io.ClusteringMongoReader;
+import uk.ac.ebi.eva.accession.clustering.batch.io.SplitOrMergedRSReader;
 import uk.ac.ebi.eva.accession.clustering.configuration.InputParametersConfiguration;
 import uk.ac.ebi.eva.accession.clustering.parameters.InputParameters;
 import uk.ac.ebi.eva.accession.core.configuration.nonhuman.MongoConfiguration;
+import uk.ac.ebi.eva.accession.core.service.nonhuman.ClusteredVariantAccessioningService;
+import uk.ac.ebi.eva.accession.core.service.nonhuman.SubmittedVariantAccessioningService;
 
-import static uk.ac.ebi.eva.accession.clustering.configuration.BeanNames.BACK_PROPAGATED_RS_READER;
+import static uk.ac.ebi.eva.accession.clustering.configuration.BeanNames.BACK_PROPAGATED_NEW_RS_TARGET_READER;
+import static uk.ac.ebi.eva.accession.clustering.configuration.BeanNames.SPLIT_OR_MERGED_RS_READER;
 
 @Configuration
 @Import({MongoConfiguration.class, InputParametersConfiguration.class})
-public class BackPropagatedRSReaderConfiguration {
+public class BackPropagatedRSTargetReaderConfiguration {
 
-    @Bean(BACK_PROPAGATED_RS_READER)
-    public ClusteringMongoReader backPropagatedRSReader(MongoTemplate mongoTemplate, InputParameters parameters) {
+    @Bean(BACK_PROPAGATED_NEW_RS_TARGET_READER)
+    public ClusteringMongoReader backPropagatedNewRSTargetReader(MongoTemplate mongoTemplate,
+                                                                 InputParameters parameters) {
         String remappedFromAssembly = parameters.getRemappedFrom();
         if (remappedFromAssembly == null) {
-            throw new IllegalArgumentException("Assembly remapped from attribute must be provided!");
+            throw new IllegalArgumentException("Assembly remappedFrom attribute must be provided!");
         }
         return new ClusteringMongoReader(mongoTemplate, remappedFromAssembly, parameters.getChunkSize(), false);
```

Comment:
If we use the `ClusteringMongoReader` we're going to get all the variants that have no `rs` fields but we should also filter out all the one that have no `backPropagatedRS`.
This will also make the process idempotent if I'm not mistaken.

Reply:
Done